src/fault_injector/fault_types/train_speed_fault.py

```diff
@@ -39,6 +40,11 @@ def resolve_fault(self, tick: int):
         :param tick: the simulation tick in which resolve_fault was called
         :type tick: Integer
         """
+        if self.train is None or self.train is not self.get_train(
+            self.simulation_object_updater, self.train.identifier
+        ):
+            raise ValueError("Train does not exist or fault not injected")
```

Comment:
And again, I do not like that this always causes an error. Despawned trains should be ok in my opinion.